org.lflang/src/org/lflang/generator/CppGenerator.kt

```diff
@@ -0,0 +1,263 @@
+            |if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+            |set    (CMAKE_BUILD_TYPE "$S{DEFAULT_BUILD_TYPE}" CACHE STRING "Choose the type of build." FORCE)
+            |endif()
+            |
+        ${
+            if (targetConfig.externalRuntimePath != null) """
+                |find_package(reactor-cpp PATHS ${targetConfig.externalRuntimePath}")
+            """.trimIndent()
+            else """
+                |if(NOT REACTOR_CPP_BUILD_DIR)
+                |    set(REACTOR_CPP_BUILD_DIR "" CACHE STRING "Choose the directory to build reactor-cpp in." FORCE)
+                |endif()
+                |
+                |ExternalProject_Add(dep-reactor-cpp
+                |   PREFIX "$S{REACTOR_CPP_BUILD_DIR}"
+                |   GIT_REPOSITORY "https://github.com/tud-ccc/reactor-cpp.git"
+                |   GIT_TAG "$runtimeVersion"
+                |   CMAKE_ARGS
+                |   -DCMAKE_BUILD_TYPE:STRING=$S{CMAKE_BUILD_TYPE}
+                |   -DCMAKE_INSTALL_PREFIX:PATH=$S{CMAKE_INSTALL_PREFIX}
+                |   -DCMAKE_INSTALL_BINDIR:PATH=$S{CMAKE_INSTALL_BINDIR}
+                |   -DCMAKE_CXX_COMPILER=$S{CMAKE_CXX_COMPILER}
+                |   -DREACTOR_CPP_VALIDATE=${if (targetConfig.noRuntimeValidation) "OFF" else "ON"}
+                |   -DREACTOR_CPP_TRACE=${if (targetConfig.tracing != null) "ON" else "OFF"} 
+                |   -DREACTOR_CPP_LOG_LEVEL=${targetConfig.logLevel.severity}
+                |)
+                |
+                |set(REACTOR_CPP_LIB_DIR "$S{CMAKE_INSTALL_PREFIX}/$S{CMAKE_INSTALL_LIBDIR}")
+                |set(REACTOR_CPP_BIN_DIR "$S{CMAKE_INSTALL_PREFIX}/$S{CMAKE_INSTALL_BINDIR}")
+                |set(REACTOR_CPP_LIB_NAME "$S{CMAKE_SHARED_LIBRARY_PREFIX}reactor-cpp$S{CMAKE_SHARED_LIBRARY_SUFFIX}")
+                |set(REACTOR_CPP_IMPLIB_NAME "$S{CMAKE_STATIC_LIBRARY_PREFIX}reactor-cpp$S{CMAKE_STATIC_LIBRARY_SUFFIX}")
+                |
+                |add_library(reactor-cpp SHARED IMPORTED)
+                |add_dependencies(reactor-cpp dep-reactor-cpp)
+                |if(WIN32)
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_IMPLIB "$S{REACTOR_CPP_LIB_DIR}/$S{REACTOR_CPP_IMPLIB_NAME}")
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_LOCATION "$S{REACTOR_CPP_BIN_DIR}/$S{REACTOR_CPP_LIB_NAME}")
+                |else()
+                |   set_target_properties(reactor-cpp PROPERTIES IMPORTED_LOCATION "$S{REACTOR_CPP_LIB_DIR}/$S{REACTOR_CPP_LIB_NAME}")
+                |endif()
+                |
+                |if (APPLE)
+                |   file(RELATIVE_PATH REL_LIB_PATH "$S{REACTOR_CPP_BIN_DIR}" "$S{REACTOR_CPP_LIB_DIR}")
+                |   set(CMAKE_INSTALL_RPATH "@executable_path/$S{REL_LIB_PATH}")
+                |else ()
+                |   set(CMAKE_INSTALL_RPATH "$S{REACTOR_CPP_LIB_DIR}")
+                |endif ()
+            """.trimIndent()
```

Comment:
I think this trimIndent is unnecessary, also the one in the other `if` branch

Reply:
I am not so sure about that. I added it because `trimIndent()` also removes blank lines at the beginning and end of the string. I would assume that without the call, there would be an additional empty line in the resulting string, but I did not try what happens without `trimIndent()`.

Reply:
Removing the `trimIndent()` indeed leads to insertian of an aditional blank line with some spaces in the resulting cmake file (basically line 220 gets inserted into the resulting file without rhe trimIndent). While this isn't critical, I think its cleaner with the `trimIndent()` in place.